docfx_yaml/extension.py

```diff
@@ -365,19 +382,133 @@ def _update_friendly_package_name(path):
         'langs': ['python'],
     }
 
-    # Only add summary to parts of the code that we don't get it from the monkeypatch
-    if _type == MODULE:
+    summary_info = {
+        'variables': {},  # Stores mapping of variables and its description, types
+        'returns': [],    # Stores the return info
+        'exceptions': []  # Stores the exception info
+    }
+
+    # Add extracted summary
+    if lines != []:
         lines = _resolve_reference_in_module_summary(lines)
         summary = app.docfx_transform_string('\n'.join(_refact_example_in_module_summary(lines)))
+    
+        # Extract summary info into respective sections.
         if summary:
-            datam['summary'] = summary.strip(" \n\r\r")
+            top_summary = ""
+
+            # Initialize known types needing further process.
+            var_types = {
+                ':rtype:': 'returns',
+                ':returns:': 'returns',
+                ':type': 'variables',
+                ':param': 'variables',
+                ':raises': 'exceptions',
+                ':raises:': 'exceptions'
+            }
+
+            # Trim the top summary but maintain its formatting.
+            indexes = []
+            for types in var_types:
+                index = summary.find(types)
+                if index > -1:
+                    indexes.append(index)
+
+            # If we found types needing further process, locate its index,
+            # if we found empty array for indexes, stop processing further.
+            index = min(indexes) if indexes else 0
+
+            # Store the top summary separately.
+            if index == 0:
+                top_summary = summary
+            else:
+                top_summary = summary[:index]
+                summary = summary[index:]
 
-    if args or sig:
+                # Clean the string by cleaning newlines and backlashes, then split by white space.
+                parsed_text = " ".join(filter(None, re.split('\n|  |\\\\',summary))).split(" ")
+            
+                cur_type = ''
+                words = []
+                arg_name = ''
+                index = 0
+
+                # Using counter iteration to easily extract names rather than
+                # coming up with more complicated stopping logic for each tags.
+                while index <= len(parsed_text):
```

Comment:
Overall comment: I'm surprised Sphinx doesn't handle this parsing under the hood. If it doesn't and we have to do this, that's fine. But, just commenting to confirm there isn't some sort of already-parsed representation of the comment?